### PR TITLE
EEP: The "Compact Core" Principle

### PR DESCRIPTION
**Pull request by dmikurube**
This is an "retrospective" EEP (Embulk Enhancement Proposal) for Embulk's historical design decisions, and upcoming design principle.

### CODE REVIEW

File: docs/eeps/draft-compact-core-principle.md [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+In this case, we decided to "fix" it in each plugin, not in the core.
+
+* [Do not use prefix in createTempFile, and release 0.2.3 (embulk/embulk-output-ftp#26)](https://github.com/embulk/embulk-output-ftp/pull/26)
+
+Embulk still has a motivation to provide `TempFileSpace` so that Embulk can control the temporary files, for example, cleaning up when a plugin aborts. It was unavoidable. But, this case still explains how difficult to make just a small change in the core. The core should provided just the minimum dependencies for plugins.
+
+### Case #3: Library dependencies of Jackson
+
+Embulk had contained [Jackson](https://github.com/FasterXML/jackson) libraries as dependencies directly in the core. Jackson consisted of multiple artifacts, from the core librarires, to supportive sub libraries. The Embulk core contained only the core libraries of Jackson, such as [`jackson-core`](https://github.com/FasterXML/jackson-core), [`jackson-annotations`](https://github.com/FasterXML/jackson-annotations), and [`jackson-databind`](https://github.com/FasterXML/jackson-databind).
```
Comment (hiroyuki-sato) [inline docs/eeps/draft-compact-core-principle.md:105]: librarires -> libraries. ?

**Review by hiroyuki-sato — APPROVED** [review]
LGTM👍

**Review by legiangthanh — APPROVED** [review]
LGTM

**Comment by dmikurube** [comment]
Thanks! As long as fixing a typo, I fixed some English phrasing in c0d1bdcf416832cf0afca15afa6467fba85b2c13, but the meanings are kept.

**Comment by dmikurube** [comment]
Let's merge it as EEP-e as `Final`.

**Pull request merged**